MP3 Tag Editor/Mp3Manager.cs

```diff
@@ -1,9 +1,221 @@
+        }
+        else
+        {
+            return ModifyNonListMp3Tags(selectedProperty, newValue);
+        }
+    }
+
+    /// <summary>
+    /// Modifies the inputted property of the working <see cref="Mp3"/> file.
+    /// Applies to "Artists" and "Composers" properties.
+    /// </summary>
+    /// <param name="selectedProperty">The <see cref="Id3Tag"/> property to be changed.</param>
+    /// <param name="newValue">The collection of values to assign to the selectedProperty.</param>
+    /// /// <returns>
+    /// An array of <see cref="Id3Tag"/>s corresponding to the input <see cref="Mp3"/>s.
+    /// </returns>
+    private static IEnumerable<Id3Tag> ModifyListMp3Tags(string selectedProperty, IEnumerable<string> newValue)
     {
-        
+        Id3Tag[] newTags = new Id3Tag[_workingMp3Files.Count];
+        newValue = newValue.ToArray(); //prevent multiple enumeration
+
+        foreach (var mp3 in _workingMp3Files)
+        {
+            var newTag = mp3.GetTag(GetTagFamily(mp3));
+
+            switch (selectedProperty)
+            {
+                case "Artists":
+                case "Artist":
+                    foreach (var value in newValue)
+                    {
+                        newTag.Artists.Value.Add(value);
+                    }
+                    break;
+
+                case "Composers":
+                case "Composer":
+                    foreach (var value in newValue)
+                    {
+                        newTag.Composers.Value.Add(value);
+                    }
+                    break;
+
+                default:
+                    throw new InvalidId3TagException("Unknown error occured!");
+            }
+
+            newTags[_workingMp3Files.IndexOf(mp3)] = newTag;
+        }
+        return newTags;
+    }
+
+    /// <summary>
+    /// Modifies the inputted property of the working <see cref="Mp3"/> file.
+    /// Applies to most properties.
+    /// </summary>
+    /// <param name="selectedProperty">The <see cref="Id3Tag"/> property to be changed.</param>
+    /// <param name="newValue">The value to assign to the selectedProperty.</param>
+    /// <returns>
+    /// An array of <see cref="Id3Tag"/>s corresponding to the input <see cref="Mp3"/>s.
+    /// </returns>
+    private static IEnumerable<Id3Tag> ModifyNonListMp3Tags(string selectedProperty, dynamic newValue)
+    {
+        Id3Tag[] newTags = new Id3Tag[_workingMp3Files.Count];
+        foreach (var mp3 in _workingMp3Files)
+        {
+            var newTag = mp3.GetTag(GetTagFamily(mp3));
+
+            //Uses Reflection to dynamically change property instead of switch statement
+            //selectedProperty must be exactly equal property in Id3Tag
+            try
+            {
+                var tagPropertyInfo =
+                    newTag.GetType().GetProperty(selectedProperty);
+
+                var tagPropertyFrame =
+                    Activator.CreateInstance(tagPropertyInfo.PropertyType, newValue);
```

Comment:
My IDE gives a null reference warning. Ignore this comment if you're sure it will never be null.
Otherwise, either check if null before or use the conditional access operator '?' as follows:
![rider64_wTzh8fdJiF](https://github.com/ZiadElGendy/mp3-tag-editor/assets/90088999/a84e8575-c337-495b-a348-78118c14bf71)

Reply:
From what I understand if these refrences are null, that would mean the input is invalid.
So I think it would be better if we just handle that as an Exception if for some reason input was not validated properly.